lib/challenge.js

```diff
@@ -61,6 +110,17 @@ var some_pet;
 var Pet;
 // Answer Starts Here
 
+var Pet = {
+  name: 'Claire',
+  species: 'dog',
+  breed: 'Rottweiler',
+  noise: 'Bark',
+
+makeNoise: function (){
+  return noise;
+}
```

Comment:
No, we don't want to do this because it would create a brand new function (i.e. Function object) every time we ran the constructor. That's wasteful, and could consume a lot of memory; it would be better to have one place to keep all our methods, so that they don't get created each time.

That's where `prototype` comes in. A Constructor's `.prototype` property is an object where you can keep all methods that you want instances (i.e. the output of your Constructor) to be able to access, without each having their own individual copy.

It's written like this:
```javascript
Pet.prototype.makeNoise = function(){
  return this.noise;
};
```